### DISCUSSION TITLE
Project status

### DISCUSSION THREAD

**Discussion by Srinuyadav149** [termux/termux-gui#61] [General]
Is this project still in active development, or has it been discontinued?

**Comment by tareksander** [comment]
I don't have time to work on it currently, but it should be mostly stable and usable by now.

**Reply by Srinuyadav149** [reply]
It is usable when termux is downloaded from GitHub. But failing when downloaded from play store. Might be signature issue.

**Reply by twaik** [reply]
We do not maintain GP version. Also it has different signature.

<details>

 Termux on the Play Store is a fork of an old version with additional, unmerged changes, and is not controlled by the dev team at https://github.com/termux. Despite its existence and version number, it is not the newest or latest version, and is significantly less functional than either the current F-Droid release or Github builds. More detail: https://github.com/termux/termux-app/discussions/4000

Unless you know exactly what you're doing, you should not use the Play Store fork. Install the suggested release from F-Droid :fdroid:.
Direct link: https://f-droid.org/en/packages/com.termux

If you have currently have the Play Store fork installed and want to install an official build from F-Droid or Github, you must uninstall Termux and all companion apps first.
That means uninstalling the main app, as well as Termux:API, :Styling, :Widget, :Task, :Float, and :Boot. Termux:X11 works differently and can be left installed.
If you need help with backing up your files: https://wiki.termux.com/wiki/Backing_up_Termux

</details>

**Comment by brandyn** [comment]
I just did this: https://sifter.org/~simon/journal/20250329.html

The one sore spot is the startup time is ~7 seconds. Any idea why that would be? If it's fixable, I am game to do the work and submit a patch.

Also, I note this in the docs: "Due to Android limitations, methods that return a value fail when the Activity isn't visible"

I suspect this is behind some issues I'm having... Likewise, is there a work-around I could help with?

**Reply by tareksander** [reply]
`termux-am` should help with the startup time a bit. That should work out of the box in the GitHub Termux version. The reason methods with return values aren't queued is to prevent the client from blocking, since they can only be run when the Activity is visible again. The queueing is a quirk I put into the json protocol, but for the protobuf protocol I changed my mind and put the responsibility of respecting the Activity lifecycle fully on the client.

**Reply by brandyn** [reply]
Ah.  From that lead, I looked into it and the termux-am call in (python) Connection() is timing out and falling back to stock am (so my startup time is slow stock am PLUS 5 second timeout on failed termux-am).  Seems to be termux-am-socket failing (Could not connect to socket: No such file or directory).  Sounds like a semi-known issue since android 12, will pursue...

**Reply by agnostic-apollo** [reply]
Its only still available in github action builds, not github releases, but should be available in next 0.119.0 beta.

**Reply by tareksander** [reply]
AFAIK `termux-am` should fail instantly, so the delay you're experiencing is the combined time of `am` and the plugin startup.

**Reply by brandyn** [reply]
It does fail instantly, but the connection.py code ignores the failure (no check for exit status) and then waits for 5 seconds for it to connect to the sockets before timing out and retrying with am.  As a quick fix I just commented out the run() of termux-am and then immediately raise timeout just inside the first try block in order to move straight to the fallback, and this has greatly improved the startup time.  (I did it that way so it's easy to revert.)

**Reply by tareksander** [reply]
Oh, seems I forgot to do the proper fallback. The proper way would be to check `.returncode` from run and skip straight ahead if the exit code wasn't 0.

**Reply by brandyn** [reply]
Is there a way to bring an already-running termux-gui applcation to the foreground (i.e., when I tap the shortcut on the homescreen, have it bring the existing one up rather than starting a new instance)?   [I've spent the last few hours looking and trying things to no avail...]

**Reply by tareksander** [reply]
> Is there a way to bring an already-running termux-gui applcation to the foreground

The `bringTaskToFront` method is what you're looking for.
